smtpmail/backend/plugin/src/main/kotlin/com/ritense/valtimo/backend/plugin/client/SmtpMailClient.kt

```
@@ -0,0 +1,100 @@
+ * limitations under the License.
+ */
+
+package com.ritense.valtimo.backend.plugin.client
+
+import com.ritense.plugin.service.PluginService
+import com.ritense.resource.service.TemporaryResourceStorageService
+import com.ritense.valtimo.backend.plugin.dto.SmtpMailContentDto
+import com.ritense.valtimo.backend.plugin.dto.SmtpMailContextDto
+import com.ritense.valtimo.backend.plugin.dto.SmtpMailPluginPropertyDto
+import com.ritense.valtimo.backend.plugin.plugin.SmtpMailPlugin
+import mu.KotlinLogging
+import org.springframework.mail.javamail.JavaMailSender
+import org.springframework.mail.javamail.JavaMailSenderImpl
+import org.springframework.mail.javamail.MimeMessageHelper
+import javax.mail.internet.MimeMessage
+
+class SmtpMailClient(
+    private val pluginService: PluginService,
+    private val storageService: TemporaryResourceStorageService
+) {
+
+    fun sendEmail(
+        mailContext: SmtpMailContextDto,
+        mailContent: SmtpMailContentDto
+    ) {
+        try {
+            val javaMailSender = javaMailSender()
+
+            val message: MimeMessage = javaMailSender.createMimeMessage()
+
+            with(MimeMessageHelper(message, true)) {
+                setFrom(mailContext.sender.address)
+                mailContext.recipients.forEach { addTo(it.address) }
+                mailContext.ccList.forEach { addCc(it.address) }
+                mailContext.bccList.forEach { addBcc(it.address) }
+                setSubject(mailContext.subject)
+                setText(mailContent.mailMessage, true)
+                mailContent.attachments.forEach {
+                    addAttachment(it.fileName) { storageService.getResourceContentAsInputStream(it.fileResourceId) }
+                }
+
+            javaMailSender.send(message)
+            }
+        } catch (e: Exception) {
+            logger.warn {
+                "Sending email with subject ${mailContext.subject} has failed with error message ${e.message}"
+            }
```

If its like this the error will be ignored and the service-task that should send the email will succeed even thought no mail was send.

---

We (with Thomas and Frits) discussed that error handling is out of scope, since it is not an option to catch this in the process. So the alternative is crashing the process. Is that preferable? Or should we let this up to the implementation?

---

Crashing the process is preferable. Because there must be a way to guarantee that an email has been send. It's then Valtimo's job to make it possible to catch the error with a 'Error Boundary Event' which doesn't work right now.